src/utils/object.ts

```diff
@@ -0,0 +1,27 @@
+/********************************************************************************
+ * Copyright (c) 2017-2021 TypeFox and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+
+ export function isNotNullish(data: unknown): data is Record<PropertyKey, unknown> {
+    return data !== null && data !== undefined;
+}
+
+export function hasOwnProperty<K extends PropertyKey>(arg: unknown, key: K): arg is Record<K, unknown> {
```

Comment:
Many usages of this function include checking the type of the property. Would it be cool to add an optional third parameter `type`? Then you could even change the `unknown` in the return type to something more specific. Here's a proposal:

```typescript
export type TypeOf<T> =
    T extends number ? 'number'
    : T extends string ? 'string'
    : T extends boolean ? 'boolean'
    : T extends bigint ? 'bigint'
    : T extends symbol ? 'symbol'
    : T extends Function ? 'function'
    : T extends object ? 'object'
    : 'undefined';

export function hasOwnProperty<K extends PropertyKey, T>(arg: unknown, key: K, type?: TypeOf<T> | ((v: unknown) => v is T)): arg is Record<K, T> {
    if (!(isObject(arg) && Object.prototype.hasOwnProperty.call(arg, key))) {
        return false;
    }
    if (typeof type === 'string') {
        return typeof arg[key] === type;
    }
    if (typeof type === 'function') {
        return type(arg[key]);
    }
    return true;
}
```

Reply:
I've incorporated it. The caller's code becomes clearer.